app/controllers/profile_controller.ts

```diff
@@ -0,0 +1,215 @@
+import type { HttpContext } from '@adonisjs/core/http'
+import {
+  loadUserWithTwoFactor,
+  userHasWebauthnCredentials,
+  markSecurityConfirmed,
+  isSecurityConfirmed,
+  SECURITY_CONFIRMATION_CHALLENGE_KEY,
+  parseTransports,
+  parseRecoveryCodes,
+  generateAndStoreTwoFactorSecret,
+} from '#services/two_factor'
+import { getRpId, getOrigin, fromBase64Url } from '#services/webauthn_service'
+import { confirmSecurityValidator } from '#validators/profile_validator'
+import { db } from '#services/db'
+import hash from '@adonisjs/core/services/hash'
+import { webauthnServer } from '#services/webauthn_server'
+import type { AuthenticationResponseJSON } from '@simplewebauthn/types'
+
+export default class ProfileController {
+  async show({ auth, inertia }: HttpContext) {
+    const user = await loadUserWithTwoFactor(auth.user!.id)
```

Comment:
Isn't the user already loaded with all the necessary fields by the `SessionKyselyUserProvider`?
I'm thinking if the `loadUserWithTwoFactor` method is even necessary.